### PR TITLE
Added bigint (Int64) support.

### PR DESCRIPTION
**Pull request by yuyaryshev**
Fixed Lint, errors, added tests.
Tests fail for node 8 and 6 because they don't have Int64.
I don't know how to fix tests for them.

Users shouldn't use Int64 on thouse nodejs versions

### CODE REVIEW

**Comment by mcollina** [comment]
Can you please lint your code? CI is failing. Also, can you add a unit test?

**Comment by yuyaryshev** [comment]
> Can you please lint your code? CI is failing. Also, can you add a unit test?

Fixed Lint, errors, added tests.
Tests fail for node 8 and 6 because they don't have Int64.
I don't know how to fix tests for them.

Users shouldn't use Int64 on thouse nodejs versions

**Comment by piranna** [comment]
Any update on this? What's missing to be merged? If it's about tests not passing on Node.js 6 and 8 because they doesn't have BigInt support, I find It totally acceptable to release a Major version and deprecate them, they are outdated anyway...

File: test/test.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ const path = require('path')
 const BufferList = require('../')
 const { Buffer } = require('buffer')
 
+function tempPath (filename) {
```
Comment (rvagg) [inline test/test.js:10]: maybe just use `os.tmpdir()` to make this work nicely

**Comment by rvagg** [comment]
Suggestions:

1. fix up that tmp dir thing using `os.tmpdir()`
2. Squash the commits focusing on this feature with a proper conventional commits style, e.g. `feat: added BigInt (Int64) support`.
3. Remove <= Node.js 12 from CI, IMO we should do 14, 16, 18, and `lts/*` + `current/*` (off the top of my head those are the correct aliases, they're a repeat of `18` but they're useful when the universe moves on and we forget to bump versions). Make the commit for this separate, `chore!: remove Node.js <= 12 support` - the `!` is important because we want to make this a semver-major release since we're dropping Node.js 12 support.

**Comment by piranna** [comment]
@rvagg in case @yuyaryshev don't provide the changes in a timely fashion, would you accept a new PR from me with them?

**Comment by rvagg** [comment]
yep, but it'd be appropriate to keep attribution of @yuyaryshev's commit(s) in the process though, so pull them in & squash

File: test/test.js (hunk 1 above)
Comment (piranna) [reply test/test.js:10]: done

**Comment by piranna** [comment]
> yep, but it'd be appropriate to keep attribution of @yuyaryshev's commit(s) in the process though, so pull them in & squash

My updated PR with your suggestions is at https://github.com/rvagg/bl/pull/109, I have preserver @yuyaryshev commits. Can you review, aprove and merge it? :-)

**Comment by rvagg** [comment]
closed by #114, finally released in v6.0.0 https://github.com/rvagg/bl/releases/tag/v6.0.0

thanks for your patience!

**Comment by piranna** [comment]
> thanks for your patience!

Welcome :-)

**Pull request closed without merging**